### PR TITLE
Update the client to allow for retrieving the `latest` alias

### PR DESCRIPTION
**Pull request by kriscon-db**
### Related Issues/PRs

 #xxx

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [x] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: tests/utils/test_validation.py [hunk 1]
```diff
@@ -165,6 +167,14 @@ def test_validate_model_alias_name_bad(alias_name):
         _validate_model_alias_name(alias_name)
     assert e.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
 
+@pytest.mark.parametrize("alias_name", BAD_ALIAS_NAMES_WITHOUT_LATEST)
```
Comment (kriscon-db) [inline tests/utils/test_validation.py:170]: Update this to just be BAD_ALIAS_NAMES

File: tests/utils/test_validation.py [hunk 2]
```diff
@@ -76,12 +76,14 @@
     "a" * 256,
     None,
     "$dgs",
-    "latest",
-    "Latest",
     "v123",
     "V1",
 ]
 
+LATEST_TESTS = ["latest", "Latest"]
```
[the remark this answers is not in the document]
Comment (kriscon-db) [reply tests/utils/test_validation.py:83]: Sure.  I can add.

File: mlflow/store/model_registry/file_store.py [hunk 3]
```diff
@@ -942,7 +942,7 @@ def delete_model_version_tag(self, name, version, key):
 
     def _get_registered_model_alias_path(self, name, alias):
         _validate_model_name(name)
-        _validate_model_alias_name(alias)
+        _validate_model_alias_name(alias, perform_latest_check=False)
```
Comment (smurching) [inline mlflow/store/model_registry/file_store.py:945]: From offline discussion, we don't want to remove validation from the OSS storage backends (file/SQL store)

File: mlflow/tracking/client.py [hunk 4]
```diff
@@ -4729,5 +4729,5 @@ def print_model_version_info(mv):
             Aliases: ["test-alias"]
         """
         _validate_model_name(name)
-        _validate_model_alias_name(alias)
+        _validate_model_alias_name(alias, perform_latest_check=False)
```
Comment (smurching) [inline mlflow/tracking/client.py:4732]: It does make sense to remove validation from this client codepath (and in `rest_store` etc if we have validation there). I think on reads just removing validation entirely is fine, we can let the backend throw on an invalid name etc. For removing client-side write validation, we should make sure analogous logic is present in all backends we maintain

File: mlflow/utils/validation.py [hunk 5]
```diff
@@ -397,7 +397,7 @@ def _validate_model_version(model_version):
         )
 
 
-def _validate_model_alias_name(model_alias_name):
+def _validate_model_alias_name(model_alias_name, perform_latest_check=True):
```
Comment (smurching) [inline mlflow/utils/validation.py:400]: We can probably get rid of this boolean / `latest`-specific logic if we switch to removing validation on reads + enforcing it on writes. And then this PR can be about removing validation on fetching model versions by alias

**Review by smurching — APPROVED** [review]

**Comment by kriscon-db** [comment]
@mlflow-automation autoformat

**Review by BenWilson2 — APPROVED** [review]

**Pull request merged**